packages/manager/modules/pci/src/projects/project/storages/databases/fork/fork.controller.js

```diff
@@ -0,0 +1,260 @@
+import find from 'lodash/find';
+import filter from 'lodash/filter';
+import get from 'lodash/get';
+import map from 'lodash/map';
+import clone from 'lodash/clone';
+import sortBy from 'lodash/sortBy';
+
+import { NAME_PATTERN } from './fork.constants';
+
+export default class {
+  /* @ngInject */
+  constructor(
+    $anchorScroll,
+    $translate,
+    $q,
+    $scope,
+    CucCloudMessage,
+    DatabaseService,
+    Poller,
+  ) {
+    this.$anchorScroll = $anchorScroll;
+    this.$translate = $translate;
+    this.$q = $q;
+    this.$scope = $scope;
+    this.CucCloudMessage = CucCloudMessage;
+    this.DatabaseService = DatabaseService;
+    this.Poller = Poller;
+    this.NAME_PATTERN = NAME_PATTERN;
+  }
+
+  $onInit() {
+    this.labAccepted = this.lab.isActivated();
+    this.messageContainer = 'pci.projects.project.storages.databases.fork';
+    this.loadMessages();
+
+    // Retrieve data from db object
+    const engine = find(this.engines, (e) => e.name === this.database.engine);
```

Comment:
Can't we use `this.database.getEngineFromList(this.engines);` ? Same applies for selectedVersion, plan, region and flavor below (can we use the existing methods in the respective classes)?

Reply:
Good point ! Done here: https://github.com/ovh/manager/pull/5431/commits/55cef43cd6973f007c3631bc8cb1ccc9bcda1132